### PR TITLE
swarm: Use pseudo-random instead of crypto-random for test files content generation

### PR DESCRIPTION
**Pull request by nizsheanez**
Related to CI speedup: https://github.com/ethereum/go-ethereum/pull/17854

Changes has been reviewed in and closes https://github.com/ethersphere/go-ethereum/pull/959

In this PR:
- Replace "crypto/rand" by "math/rand" for files content generation
- Use testutil.TempFileWithContent where possible
- Move testutil/http.go file to swarm/api/http package because it depends on swarm/api and swarm/storage packages (it means that testutil packager not possible to use from that packages - cyclic imports)
- Use math/rand.Shuffle instead of swarm/network_test.go.Shuffle and swarm/btm/btm_test.go.Shuffle. Because math/rand.Shuffle in stdlib since go1.10 (go1.9 support dropped in: https://github.com/ethereum/go-ethereum/pull/17807)

### CODE REVIEW

File: swarm/network/stream/delivery_test.go [hunk 1]
```diff
@@ -530,7 +529,8 @@ func testDeliveryFromNodes(t *testing.T, nodes, conns, chunkCount int, skipCheck
 		//now we can actually upload a (random) file to the round-robin store
 		size := chunkCount * chunkSize
 		log.Debug("Storing data to file store")
-		fileHash, wait, err := roundRobinFileStore.Store(ctx, io.LimitReader(crand.Reader, int64(size)), int64(size), false)
+		b := testutil.RandomBytes(1, size)
+		fileHash, wait, err := roundRobinFileStore.Store(ctx, bytes.NewReader(b), int64(size), false)
```
Comment (nolash) [inline swarm/network/stream/delivery_test.go]: Why not use `testutil.RandomReader()` in this case? (there are more of these cases further down the diff)

Comment (zelig) [reply swarm/network/stream/delivery_test.go]: true

File: cmd/swarm/upload_test.go [hunk 2]
```diff
@@ -77,33 +78,22 @@ func testCLISwarmUp(toEncrypt bool, t *testing.T) {
...
 	defer cluster.Shutdown()
 
-	// create a tmp file
-	tmp, err := ioutil.TempFile("", "swarm-test")
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer tmp.Close()
-	defer os.Remove(tmp.Name())
+	tmpFileName := testutil.TempFileWithContent(t, data)
```
[the remark this answers is not in the document]
Comment (nizsheanez) [reply cmd/swarm/upload_test.go]: No. There is existing `data` constant in `cmd/swarm/access_test.go` - `data` variable was shadowed in this function, but had same value. I cleanup couple more places to reuse that constant.

File: swarm/network/stream/delivery_test.go (hunk 1 above)
Comment (nizsheanez) [reply swarm/network/stream/delivery_test.go]: Fixed everywhere except. `swarm/storage/filestore_test.go:testFileStoreRandom` - this test looks a bit weird for me. `newTestDbStore` using `ioutil.TempDir` inside, but test cleanup `/tmp` folder. Also i'm not sure why test writing files like `/tmp/slice.bzz.16M`. Looks like this test need to cleanup, I prefer to do it separately.

Comment (nolash) [reply swarm/network/stream/delivery_test.go]: you mean `tmp/bzz`? I agree, there's something not too obvious about that test. And yes, it leaves behind garbage. Please fix. I'm fine with separate PR for that.

**Pull request merged**